modules/live-schedule-impl/src/main/java/org/opencastproject/liveschedule/impl/LiveScheduleServiceImpl.java

```diff
@@ -257,6 +261,12 @@ protected void activate(ComponentContext context) {
 
     if (!StringUtils.isBlank((String) properties.get(LIVE_DISTRIBUTION_SERVICE))) {
       distributionServiceType = StringUtils.trimToEmpty((String) properties.get(LIVE_DISTRIBUTION_SERVICE));
+    } else {
+      distributionServiceType = DEFAULT_LIVE_DISTRIBUTION_SERVICE;
+    }
+    if (tempDistributionServices.get(distributionServiceType) != null) {
```

Comment:
The issue with this approach is that in a misconfiguration scenario (ie, I say I want a dist service which does not exist on this node), you'll have a Karaf service which *looks* like it's working but will NPE when you go to use it - since nothing has set `downloadDistributionService`.

Take a look at `etc/org.opencastproject.speechtotext.impl.SpeechToTextServiceImpl.cfg` and `modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/SpeechToTextServiceImpl.java`, which do *exactly* what you're trying to do with via `SpeechToTextEngine.target`, which set the OSGi DS filter at boot ;)

Reply:
@gregorydlogan , the idea is great, but I don't think that example is working. It's true that there's a configuration, but the code does not use it and the reference in that example is static so when the component is up the reference is already set. Do you have another example that works?

Reply:
Hi @gregorydlogan , thanks for the suggestion! I think it's much better now!